### PR TITLE
adding initial redis output support

### PR DESCRIPTION
**Pull request by dvas0004**
Adding initial support for a "redis list" output. Tested and working fine on ubuntu 22 LTS 

Documentation reproduced below for your convenience as it pretty much sums it up

### Redis

This output type sends events to a Redis list using the [LPUSH command](https://redis.io/commands/lpush/)

You must provide:
- a redis server address containing the IP and port to connect to.
- a list name

TODO:

- [ ] implement TLS connections to redis
- [ ] support redis auth
- [ ] ...

#### Examples

* Send events to a redis server into a list named "wec":

```
$ openwec subscriptions edit my-test-subscription outputs add --format <format> redis 127.0.0.1:6377 wec
```

### CODE REVIEW

**Comment by vruello** [comment]
Thank you for this nice contribution :+1: 

I see that you updated the PR this morning, is it ready to be reviewed? Do you plan to implement the TODOs you wrote now or do you prefer the initial support to be merged before?

**Comment by dvas0004** [comment]
hey @vruello !!

to be honest i would go ahead and review it, the TODOs will come but at the moment they are "nice to haves" and the system is already working very well without them

**Review by vruello — CHANGES_REQUESTED** [review]
I've made some suggestions for improvement! Btw, I really appreciate your efforts to document the output usage :+1: 

Could you add the feature in `CHANGELOG.md`?

File: server/src/outputs/redis.rs [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+pub struct OutputRedis {
+    format: Format,
+    config: RedisConfiguration,
+    producer: redis::Client,
+}
+
+impl OutputRedis {
+    pub fn new(format: Format, config: &RedisConfiguration) -> Result<Self> {
+
+        let client = redis::Client::open(format!("redis://{}/", config.addr())).unwrap();
```
[the remark this answers is not in the document]
Comment (dvas0004) [reply server/src/outputs/redis.rs:20]: 💯

File: server/src/outputs/redis.rs [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+                redis_cmd
+                    .arg(&[self.config.list(), event.as_ref()])
+                    .query_async::<_, Option<u32>>(&mut redis_connection)
+                    .await
+            });
+
+        }
+
+        while let Some(result) = results.next().await {
+            debug!("Redis message sent: {:?}", result)
```
[the remark this answers is not in the document]
Comment (dvas0004) [reply server/src/outputs/redis.rs:61]: agreed... i'll base it off the Kafka output where you've done this...

**Comment by dvas0004** [comment]
thank you very much for your feedback @vruello !! i just committed your feedback👍

File: .idea/.gitignore
[the remark this answers is not in the document]
Comment (vruello) [reply .idea/.gitignore:1]: You made the addition in `.gitignore`, but you did not remove the previously committed files in `.idea/`. Could you do this? (you can see on https://github.com/cea-sec/openwec/pull/45/files that they are still there)

File: server/src/outputs/redis.rs [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+                redis_cmd
+                    .arg(&[self.config.list(), event.as_ref()])
+                    .query_async::<_, Option<u32>>(&mut redis_connection)
+                    .await
+            });
+
+        }
+
+        while let Some(result) = results.next().await {
+            debug!("Redis message sent: {:?}", result);
```
Comment (vruello) [inline server/src/outputs/redis.rs:61]: This line should be removed since the `debug` is in the `match`.

**Review by vruello — APPROVED** [review]

**Comment by vruello** [comment]
Once again, thank you very much for your contribution :+1:

**Pull request merged**